File: app/Models/Topic.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

/**
 * App\Models\Topic
 *
 * @property int $id
 * @property int $user_id
 * @property int $forum_id
 * @property string $subject 主题标题
 * @property string $content 主题内容
 * @property int $view_count 查看次数
 * @property int $position 置顶位置
 * @property int $is_hot 是否热门
 * @property \Illuminate\Support\Carbon|null $created_at
 * @property \Illuminate\Support\Carbon|null $updated_at
 * @property \Illuminate\Support\Carbon|null $deleted_at
 * @method static bool|null forceDelete()
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic newModelQuery()
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic newQuery()
 * @method static \Illuminate\Database\Query\Builder|\App\Models\Topic onlyTrashed()
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic query()
 * @method static bool|null restore()
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic whereContent($value)
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic whereCreatedAt($value)
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic whereDeletedAt($value)
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic whereForumId($value)
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic whereId($value)
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic whereIsHot($value)
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic wherePosition($value)
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic whereSubject($value)
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic whereUpdatedAt($value)
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic whereUserId($value)
 * @method static \Illuminate\Database\Eloquent\Builder|\App\Models\Topic whereViewCount($value)
 * @method static \Illuminate\Database\Query\Builder|\App\Models\Topic withTrashed()
 * @method static \Illuminate\Database\Query\Builder|\App\Models\Topic withoutTrashed()
 * @mixin \Eloquent
 */
class Topic extends Model
{
    use SoftDeletes;
}

File: app/controller/response/topic/topic.go
package topic

type Topic struct {
	ID        int         `json:"id" example:"1"`                      // ID
	User      interface{} `json:"user"`                                // 用户
	Forum     interface{} `json:"forum"`                               // 论坛版块
	Title     string      `json:"title" example:"This is a title"`     // 主题标题
	Content   string      `json:"content" example:"This is a content"` // 主题内容
	ViewCount int         `json:"view_count" example:"0"`              // 查看次数
	Position  int         `json:"position" example:"0"`                // 置顶位置
	IsHot     bool        `json:"is_hot" example:"false"`              // 是否热门
	CreatedAt int         `json:"created_at" example:"1591974665"`     // 创建时间戳（秒）
	UpdatedAt int         `json:"updated_at" example:"1591974665"`     // 更新时间戳（秒）
}

type User struct {
	ID          int    `json:"id" example:"1"`                // 用户ID
	DisplayName string `json:"display_name" example:"spider"` // 用户名
}

type Forum struct {
	ID   int    `json:"id" example:"1"`                      // 版块ID
	Name string `json:"name" example:"This is a forum name"` // 版块名称
}
